### PR TITLE
Block people from upgrading Airflow in the image

### PR DESCRIPTION
**Pull request by ashb**
If they do this it will break various things, so we don't want to let
them do this

### CODE REVIEW

File: .circleci/bin/test-airflow-image.py [hunk 1]
```diff
@@ -171,6 +171,18 @@ def test_airflow_trigger_dags(scheduler):
...
+def test_apache_airflow_in_requirements(tmp_path):
+    """Add test to check that docker build errors when apache-airflow is specified in requirements.txt"""
+    test_project = tmp_path / "test_project"
+    test_project.mkdir()
+
+    (test_project / "Dockerfile").write_text(f"FROM {os.environ['AIRFLOW_IMAGE']}")
+    (test_project / "requirements.txt").write_text("apache-airflow")
+    (test_project / "packages.txt").write_text("")
+    output = subprocess.run(['docker', 'build', '-t', 'testimage', test_project.resolve()], capture_output=True)
+    assert b"Do not upgrade by specifying 'apache-airflow' in your requirements.txt" in output.stderr
```
[the remark this answers is not in the document]
Comment (kaxil) [reply .circleci/bin/test-airflow-image.py:186]: I will need to rebase this PR once https://github.com/astronomer/ap-airflow/pull/99 is merged

[the remark this answers is not in the document]
Comment (kaxil) [reply .circleci/bin/test-airflow-image.py:201]: Rebased

**Comment by kaxil** [comment]
> Huh, curious, someone has submitted a patch for the buster vuln https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=962237 - it just hasn't been released.

The notes at https://security-tracker.debian.org/tracker/CVE-2020-10878 says it is a Minor issue. for buster & stretch
image

**Comment by ashb** [comment]
Turns out this broke a valid use case -- installing extras :

```
apache-airflow[hive]
```

We should probably re-enable that use-case/behaviour.

**Pull request merged**